src/utils/scval.ts

```diff
@@ -2,6 +2,9 @@ export function scaleInputToBigInt(input: string, decimals: number): bigint {
   let scaled_input;
   if (input.includes('.')) {
     let [base, decimal] = input.split('.');
+    if (decimal.length > decimals) {
```

Comment:
why return 0? wouldn't it be better to round to the specific decimals ?

Reply:
also curious about this edge case? This seems like it should do:

decimals: 3, input "1.1234"

expected -> BigInt(1123)
actual -> BigInt(0)

Reply:
I added this because the simulation for new positions is always run when the input for pool action is changed. The input error handling does not allow you to go over the decimals that the token supports for submitting a transaction. But since this does not apply to the simulation. This function when run in the simulation with an input that has more decimals than are supported  throws an error because we attempt to repeat trailing zeros by a negative number.

Reply:
We could round but that might cause transactions to fail when they shouldn't. The user should not be able to input more decimals than the token supports.

Reply:
Not sure I follow the justification.

The basic input error handling (for cleansing things like too many decimals) should be run before we attempt to simulate the transaction.

The function here should still work as expected, I think, and truncate a number with too many decimals.